File: UmarApp-UI/src/app/core/models/Sale.ts
export interface Sale {
  id: string; 
  productId: string; 
  quantity: number;
  price: number;
  totalAmount: number;
  saleDate?: string;
}

export interface SaleDto {
  productId: string;
  quantity: number;
  price: number;
  totalAmount: number;
  saleDate: string;
}

export interface SaleOutputDto {
  id: string; 
  productId: string; 
  productName: string; 
  quantity: number;
  price: number;
  totalAmount: number;
  saleDate?: string;
}

File: UmarAppApi/Models/Sales/Sale.cs
using UmarAppApi.Models.Products;

namespace UmarAppApi.Models.Sales
{
    public class Sale
    {
        public Guid Id { get; set; }
        public Guid ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public decimal TotalAmount { get; set; }
        public DateTime SaleDate { get; set; }
    }
}
